src/Selenium2Library/keywords/_screenshot.py

```diff
@@ -17,49 +17,77 @@ def __init__(self):
         """Sets the root output directory for captured screenshots.
 
-        ``path`` argument specifies the absolute path where the screenshots should
-        be written to. If the specified ``path`` does not exist, it will be created.
-        Setting ``persist`` specifies that the given ``path`` should
-        be used for the rest of the test execution, otherwise the path will be restored
-        at the end of the currently executing scope.
+        ``path`` argument specifies the absolute path where the screenshots
+        should be written to. If the specified ``path`` does not exist,
+        it will be created. Setting ``persist`` specifies that the given
+        ``path`` should be used for the rest of the test execution, otherwise
+        the path will be restored at the end of the currently executing scope.
         """
         path = os.path.abspath(path)
         self._create_directory(path)
         if persist is False:
             self._screenshot_path_stack.append(self.screenshot_root_directory)
             # Restore after current scope ends
-            utils.events.on('scope_end', 'current', self._restore_screenshot_directory)
+            utils.events.on('scope_end', 'current',
+                            self._restore_screenshot_directory)
 
         self.screenshot_root_directory = path
 
     def capture_page_screenshot(self, filename=None):
         """Takes a screenshot of the current page and embeds it into the log.
 
-        `filename` argument specifies the name of the file to write the
-        screenshot into. If no `filename` is given, the screenshot is saved into file
-        `selenium-screenshot-<counter>.png` under the directory where
-        the Robot Framework log file is written into. The `filename` is
+        ``filename`` argument specifies the name of the file to write the
+        screenshot into. If no ``filename`` is given, the screenshot is saved
+        into file `selenium-screenshot-<counter>.png` under the directory where
+        the Robot Framework log file is written into. The ``filename`` is
         also considered relative to the same directory, if it is not
         given in absolute format. If an absolute or relative path is given
         but the path does not exist it will be created.
 
-        `css` can be used to modify how the screenshot is taken. By default
-        the bakground color is changed to avoid possible problems with
-        background leaking when the page layout is somehow broken.
+        New in Selenium2Library 1.8
+
+        If ``filename`` contains %d character, it will be automatically
+        replaced with running counter. The running counter is unique for each
+        different filename. The absolute path of the saved screenshot is always
+        returned.
+
+        If ``filename`` contains two or more %d characters or there is only one
+        or more % characters in filename, then a TypeError exception is raised.
```

Comment:
It is because of Python older string formatting, which the code uses, see in here: https://github.com/robotframework/Selenium2Library/pull/504/files#diff-5346a546c6b37e3c016ad51e7155887dR122. Example this: ```python -c "'foo % bar %d' % 'string'"``` will also cause exception. There is bug in documentation, that exception is not always TypeError and it would be best to say only exception. 

The Python 2.6 onward supports ```str.format()```, which could solve the problem in different way but because we must support Python 2.5, I did not use it. I could also do simple ```str.replace()``` in the code, but I would like to keep some formatting options open for future enhancements.